os/drivers/bluetooth/virtual_ble/ble_vdev.c

```diff
@@ -19,8 +19,873 @@
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+trble_result_e vdev_stop_scan(struct bledev *dev)
+{
+    if (is_scan_started == false)
+    {
+        VBLE_LOG("Scan not running");
+        return TRBLE_INVALID_STATE;
+    }
+    trble_result_e tres = TRBLE_SUCCESS;
+    struct vble_req req = {VBLE_MSG_STOP_SCAN, NULL, &tres};
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+trble_result_e vdev_scan_whitelist_add(struct bledev *dev, trble_addr *addr)
+{
+    if (addr == NULL) {
+        return TRBLE_INVALID_ARGS;
+    } else if (is_scan_started != false) {
+        return TRBLE_INVALID_STATE;
+    }
+    trble_result_e tres = TRBLE_SUCCESS;
+    struct vble_req req = {VBLE_MSG_SCAN_WHITELIST_ADD, (void *)addr, &tres};
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+trble_result_e vdev_scan_whitelist_delete(struct bledev *dev, trble_addr *addr)
+{
+    if (addr == NULL) {
+        return TRBLE_INVALID_ARGS;
+    } else if (is_scan_started != false) {
+        return TRBLE_INVALID_STATE;
+    }
+    trble_result_e tres = TRBLE_SUCCESS;
+    struct vble_req req = {VBLE_MSG_SCAN_WHITELIST_DELETE, (void *)addr, &tres};
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+trble_result_e vdev_scan_whitelist_clear_all(struct bledev *dev)
+{
+    if (is_scan_started != false) {
+        return TRBLE_INVALID_STATE;
+    }
+    trble_result_e tres = TRBLE_SUCCESS;
+    struct vble_req req = {VBLE_MSG_SCAN_WHITELIST_CLEAR_ALL, NULL, &tres};
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+/*** Central(Client) ***/
+trble_result_e vdev_client_connect(struct bledev *dev, trble_conn_info *conn_info)
+{
+    trble_result_e tres = TRBLE_SUCCESS;
+    struct vble_req req = {VBLE_MSG_CLIENT_CONNECT, (void *)conn_info, &tres};
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+trble_result_e vdev_client_disconnect(struct bledev *dev, trble_conn_handle conn_handle)
+{
+    trble_result_e tres = TRBLE_SUCCESS;
+    struct vble_req req = {VBLE_MSG_CLIENT_DISCONNECT, (void *)&conn_handle, &tres};
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+trble_result_e vdev_client_disconnect_all(struct bledev *dev)
+{
+    trble_result_e tres = TRBLE_SUCCESS;
+    struct vble_req req = {VBLE_MSG_CLIENT_DISCONNECT_ALL, NULL, &tres};
+    int res = _progress_message(&req);
+    if (res < 0) {
+        return TRBLE_FAIL;
+    }
+    return tres;
+}
+
+trble_result_e vdev_connected_device_list(struct bledev *dev, trble_connected_list *out_connected_list)
+{
+    if (out_connected_list == NULL)
```

Comment:
Please follow the TizenRT Coding Rules like below.
`if (xxx) {`

Reply:
Done